narwhal/network/src/p2p.rs

```diff
@@ -351,15 +351,18 @@ impl UnreliableNetwork<WorkerBatchRequest> for P2pNetwork {
 impl PrimaryToWorkerRpc for P2pNetwork {
     async fn request_batch(
         &self,
-        peer: &NetworkPublicKey,
+        peer: NetworkPublicKey,
```

Comment:
Without this change the code in `block_waiter.rs:137` doesn't work because the worker name looked up from the cache is a temporary. It seemed easiest to just make the async fn own its arguments as a solution to this but if there's a better way, I'm open!